File: BackEnd/src/main/java/todo/list/domain/Card.java
package todo.list.domain;

import java.time.LocalDateTime;

public class Card {
    private Long id;
    private String title;
    private String contents;
    private CardStatus status;
    private Author author;
    private LocalDateTime updateDateTime;

    public Card(Long id, CardStatus status) {
        this(id, null, null, status, LocalDateTime.now(), null);
    }

    public Card(String title, String contents, CardStatus status, Author author) {
        this(null, title, contents, status, LocalDateTime.now(), author);
    }

    public Card(Long id, String title, String contents, Author author) {
        this(id, title, contents, null, null, author);
    }

    public Card(Long id, String title, String contents, CardStatus status, LocalDateTime updateDateTime, Author author) {
        this.id = id;
        this.title = title;
        this.contents = contents;
        this.status = status;
        this.updateDateTime = updateDateTime;
        this.author = author;
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getContents() {
        return contents;
    }

    public CardStatus getStatus() {
        return status;
    }

    public Author getAuthor() {
        return author;
    }

    public LocalDateTime getUpdateDateTime() {
        return updateDateTime;
    }
}

File: TodoList/TodoList/domain/Card.swift
//
//  Card.swift
//  TodoList
//
//  Created by 김동준 on 2022/04/05.
//

import Foundation

struct Card: Codable {
    private(set) var status: CardStatus
    private(set) var title: String
    private(set) var contents: String
    private(set) var date: Date
    private(set) var writer: String
    
    init(status: CardStatus, title: String, contents: String, writer: String) {
        self.status = status
        self.title = title
        self.contents = contents
        self.writer = writer
        self.date = Date.now
    }
}

extension Card: CardUsable {
    func getStatus() -> CardStatus {
        return status
    }

    func getTitle() -> String {
        return title
    }

    func getContents() -> String {
        return contents
    }

    func getDate() -> Date {
        return date
    }

    func getWriter() -> String {
        return writer
    }
}
